### DISCUSSION TITLE
Proxy enable but in stats don't see the app

### DISCUSSION THREAD

**Discussion by tiennlm** [engageub/InternetIncome#27] [General]
Hi engageub,

Firstly, thank for your tools.

I setup it in my ubuntu, no proxy is ok but when turn on proxy, the app don't work.

I use proxy from my own VPS,

As example: I test with traffmonetizer, proxy when run on windows is ok, but in docker inspect that not show IP

  "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {},
            "SandboxKey": "",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "",
            "Gateway": "",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "",
            "IPPrefixLen": 0,
            "IPv6Gateway": "",
            "MacAddress": "",
            "Networks": {}

Please help me from this error.

Thank you so much.

**Comment by engageub** [comment]
Hi,
Thank you for your feedback. Proxy is not configured through environment variable, so it will not be visible in docker inspect. 
Please check your traffmonetizer dashboard to check your IP address. You will see option as part of filter in dashboard to make the device IP visible.
Please make sure you enable USE_PROXIES=true in the properties.conf file.
Please enable logs in properties.conf file to debug if you have issue with your proxy setup.
After enabling the logs, please check the logs for containers to see if there are any errors. 

Thank you

**Comment by tiennlm** [comment]
Hi,
Where is the logs file?
I check on traffmonetizer dashboard, it don't show the device that I think it don't run.
I run this tool on ubuntu vnmware.

**Comment by engageub** [comment]
Could you please check docker logs using the container names.
You will find container names or ids in containernames.txt or containers.txt in the same folder where you have proxies.txt file. 

Take each ID from text file and run the following command and input your container id or container name.
You will get the logs for traffmonetizer and tun2socks. 
You will know the reason if there is any issue.

```
sudo docker container logs <container_name or container id here>
```

Thank you

**Comment by tiennlm** [comment]
Hi
The tun container got that message

time="2023-06-18T07:11:16Z" level=info msg="[DIALER] set fwmark: 0x22b"

time="2023-06-18T07:11:21Z" level=warning msg="[TCP] dial 136.243.150.101:443: HTTP auth required by proxy"

Because my proxy have username and password. How to fix it?

My proxy format protocol://username:password@ip:port

**Comment by engageub** [comment]
Is your protocol http or https or socks5?
Please note that http and socks5 have been confirmed to be working fine.
Does your username and password have special characters like : or @ or //. This might interfere with the format.
Please note that if your ISP is blocking some IPs it will not work.
Traffmonetizer was tested on different ISP's and few  of the proxies were showing 502 error in the logs. So there may be timeout issue or the ISP is blocking the IPs when connecting to traffmonetizer.
The format is correct but if there are special characters mentioned there are chances that it may not work, however since it was confirmed by you that it is working on honeygain, it might be someother reason.
But make sure you use both honeygain and other apps in the same properties file and confirm it only happens with traffmonetizer. 

Thank you

**Comment by tiennlm** [comment]
Username and password are only number and word character, don't have special character.
I change proxy to http://IP:port and it worked fine.

But if don't have username:pass, it is risky, less secure for proxy.

**Comment by engageub** [comment]
Yes, it is recommended to use username and password to keep your proxies secure. If you are doing a manual setup, make sure you follow your documentation and raise issue to necessary team if you are missing something while creating username and password and also see if there is some encoding or decoding formats which might be causing the issue.
But if it works for one application it should work for other applications since the application uses the network from tun2socks containers and they are isolated from each other. 

Thank you

**Comment by tiennlm** [comment]
I think username and password not support from tun2socks.

Remove all username and password then script work like a charm :(

**Comment by engageub** [comment]
No, it works with username and password. The way your authentication server was configured may be different than a regular configuration.
It works with username and password.
If you purchase from proxy provider there may not be much issues but if you have never setup or used a different way of authentication, there may be issues.


Thank you

**Comment by tiennlm** [comment]
I test from another proxy provider, it the same issues.
Should you suggest some proxy provider please.

I am very appreciate with your support.
Thank you so much

**Comment by engageub** [comment]
It works on other apps but not on traffmonetizer?
Are you using the latest code from github?
Can you try with socks5 proxies also. 
In the properties.conf file did you give spaces in device name?
Do you see errors in traffmonetizer logs also?
Could you please provide the proxy provide name from where you purchased?

Thank you

**Reply by engageub** [reply]
Did you buy premium proxy or Dedicated proxy of residential or you are using the free version?
Would like to know which proxies are working for honeygain.

Thank you

**Reply by engageub** [reply]
Do you have the same issue with username and password for traffmonetizer and on removing it works fine?

Thank you

**Reply by engageub** [reply]
There is no issue here with using proxy with username and password.  Was it working with previous versions or you downloaded recently.

Thank you

**Comment by tiennlm** [comment]
All app worked with http://ip:port
When add http://username:password@ip:port it error "HTTP auth required by proxy" all

I use device name default "ubuntu"
I test on webshare proxy

**Comment by engageub** [comment]
You may use IP authentication instead of username and password however there should not be problem with username and password authentication. 
Only one authentication method is allowed at a time. 
There may be bad proxies and make sure you autoreplace bad proxies.
Please note that the script is working as expected. Please see the below config for webshare. 
Make sure your concurrency limit is not reached. 


![image](https://github.com/engageub/InternetIncome/assets/103608400/5f3046ab-5a30-468b-98c5-45ff30bbe80c)


Thank you

**Comment by engageub** [comment]
Did you buy premium proxy or Dedicated proxy of residential or you are using the free version?
Would like to know which proxies are working for honeygain.

Thank you

**Reply by tiennlm** [reply]
Many thank if you share some proxies provider can work with honeygain

**Reply by engageub** [reply]
Are webshare proxies working with honeygain?

**Reply by engageub** [reply]
Could you please let us know which proxy providers are working good with honeygain.

**Comment by engageub** [comment]
After the recent update in tun2socks, http is not working but socks5 proxies are working fine. Will be switching to older version of tun2socks.
Thank you

**Reply by engageub** [reply]
It has already been updated. Please download from github again

**Comment by engageub** [comment]
Please update the code now. It has been switched to specific version of tun2socks which was working earlier.
Thank you

**Comment by engageub** [comment]
Code has already been pushed. Issue has been raised to dependent docker container.
https://github.com/xjasonlyu/tun2socks/issues/266

**Comment by engageub** [comment]
Please download the code again from github and check if its working fine now.

Thank you

**Comment by engageub** [comment]
Closing this since the issue has been identified and updated. Reopen this in case you still see any issue.
Thank you
